aws-android-sdk-cognitoidentityprovider/src/main/java/com/amazonaws/mobileconnectors/cognitoidentityprovider/CognitoUserPool.java

```diff
@@ -274,6 +274,26 @@ public CognitoUserPool(Context context, String userPoolId, String clientId, Stri
         this.pinpointEndpointId = CognitoPinpointSharedContext.getPinpointEndpoint(context, pinpointAppId);
     }
 
+    /**
+     * Constructs a user-pool with custom endpoint for Cognito User Pool
+     *
+     * @param context               REQUIRED: Android application context.
+     * @param userPoolId            REQUIRED: User-pool-Id of the user-pool.
+     * @param clientId              REQUIRED: Client-Id generated for this app and user-pool at the
+     *                              Cognito Identity Provider developer console.
+     * @param clientSecret          REQUIRED: Client Secret generated for this app and user-pool at
+     *                              the Cognito Identity Provider developer console.
+     * @param pinpointAppId         REQUIRED: AWS Pinpoint App Id for analytics.
+     * @param cognitoUserPoolCustomEndpoint REQUIRED: Custom endpoint for Cognito Userpool
+     */
+    public CognitoUserPool(Context context, String userPoolId, String clientId, String clientSecret, AmazonCognitoIdentityProvider client, String pinpointAppId, String cognitoUserPoolCustomEndpoint) {
```

Comment:
If this constructor has the largest number of arguments, it probably makes sense to have it implement all of the assignments itself.

The existing constructor (which does not have a `cognitoUserPoolCustomEndpoint`) would then call this with `null` for the newly added argument.